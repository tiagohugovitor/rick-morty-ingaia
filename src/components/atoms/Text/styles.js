import styled from 'styled-components';

const StyledText = styled.span`
  color: ${(props) => props.theme.palette.text[props.color]};
  font-size: ${(props) => props.theme.typography.size[props.size]};
  font-weight: ${(props) => props.theme.typography.weight[props.fontWeight]};
`;

export {
  StyledText,
};
