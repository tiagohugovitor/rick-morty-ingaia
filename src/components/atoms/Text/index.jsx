import React from 'react';
import PropTypes from 'prop-types';
import {StyledText} from './styles';

const Text = ({
  text,
  color,
  size,
  fontWeight,
}) => {
  return (
    <StyledText
      color={color}
      size={size}
      fontWeight={fontWeight}
    >
      {text}
    </StyledText>
  );
};

Text.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
  fontWeight: PropTypes.string,
};

Text.defaultProps = {
  text: '',
  color: 'white',
  size: 'small',
  fontWeight: 'normal',
};

export default Text;
