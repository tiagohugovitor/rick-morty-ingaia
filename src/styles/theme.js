import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  typography: {
    fontFamily: "nunito",
    size: {
      mini: '12px',
      tiny: '13px',
      small: '14px',
      mediumSmall: '16px',
      medium: '18px',
      large: '20px',
      huge: '24px',
      extra: '28px'
    },
    weight: {
      light: '300',
      normal: '400',
      bold: '700'
    }
  },
  palette: {
    primary: {
      main: "#CBD736",
      secondary: "#C9D636"
    },
    text: {
      primary: "#9F9F9F",
      secondary: "#8C8C8C",
      tertiary: "#CBD736",
      white: "#FFFFFF",
      black: "#000000",
      grey: "#D3D3D3"
    },
    warning: {
      main: "#FF0000"
    },
    white: {
      main: "#FFFFFF"
    },
    black: {
      main: "#000000"
    },
    background:{
      primary: "#1E1E1E",
      secondary: "#303030",
      tertiary: "#1B1B1B"
    },
    border: {
      primary: "#606060"
    }
  }
});

export default theme;