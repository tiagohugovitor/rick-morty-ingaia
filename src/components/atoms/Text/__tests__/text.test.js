import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import { ThemeProvider } from 'styled-components'
import theme from '../../../../styles/theme';
import Text from '../index';

configure({ adapter: new Adapter() });

describe('Render Text Component', () => {
  it('render correctly Text component', () => {
    const TextComponent = renderer.create(
      <ThemeProvider theme={theme}>
        <Text />
      </ThemeProvider>
    ).toJSON();
    expect(TextComponent).toMatchSnapshot();
  });

  it('render correctly with default value of props', () => {
    const defaultProps = {};
    const TextComponent = mount(
      <ThemeProvider theme={theme}>
        <Text {...defaultProps} />
      </ThemeProvider>
    ).find(Text).first();
    expect((TextComponent).prop('text')).toEqual('');
    expect((TextComponent).prop('color')).toEqual('white');
    expect((TextComponent).prop('size')).toEqual('small');
    expect((TextComponent).prop('fontWeight')).toEqual('normal');
  });

  it('render correctly with null values of props', () => {
    const nullProps = {
      text: null,
      color: null,
      size: null,
      fontWeight: null,
    };
    const TextComponent = mount(
      <ThemeProvider theme={theme}>
        <Text {...nullProps} />
      </ThemeProvider>
    ).find(Text).first();
    expect((TextComponent).prop('text')).toEqual(null);
    expect((TextComponent).prop('color')).toEqual(null);
    expect((TextComponent).prop('size')).toEqual(null);
    expect((TextComponent).prop('fontWeight')).toEqual(null);
  });

  it('render correctly with normal value of props', () => {
    const props = {
      text: "Testing text",
      color: "grey",
      size: "medium",
      fontWeight: "bold",
    };
    const TextComponent = mount(
      <ThemeProvider theme={theme}>
        <Text {...props} />
      </ThemeProvider>
    ).find(Text).first();
    expect((TextComponent).prop('text')).toEqual('Testing text');
    expect((TextComponent).prop('color')).toEqual('grey');
    expect((TextComponent).prop('size')).toEqual('medium');
    expect((TextComponent).prop('fontWeight')).toEqual('bold');
  });

});