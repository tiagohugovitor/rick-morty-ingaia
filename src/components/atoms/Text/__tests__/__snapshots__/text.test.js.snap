// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Text Component render correctly Text component 1`] = `
<span
  className="sc-bdvvaa hfcXKE"
  color="white"
  fontWeight="normal"
  size="small"
>
  
</span>
`;
